api/metrics/taxitrips.py

```diff
@@ -0,0 +1,35 @@
+from api.utils.database import rows_to_dicts
+
+#multiple metrics for the taxi trip dataset
+class TaxiTripMetrics:
+    def __init__(self, connection):
+        self.connection = connection
+        
+        
+#gets the average speed of a taxi 
+    def avgtaxispeed(self):
```

Comment:
This is a cool idea for a metric! Can you make it a group by pickup or dropoff community area and then rename the avg speed column as value? Then we could plug this into the community scatter view and see average speeds by community of pickup or dropoff.

Reply:
Should I still group it by taxi_id, too?

Reply:
That's a good point, I think you actually want to get average speed grouped by trip_id, not by taxi_id, and then group that by community area.

You want to take the **average average** speed of trips in each community area.

Reply:
I was thinking of making that another metric, average trip speed, because I wanted this one to be the average speed of different taxis. Unless you think average speed of taxis isn't really needed?

Reply:
That's a good idea, make them as separate metrics:

1. avg_speed_per_pickup_area()
2. avg_speed_per_dropoff_area()
3. avg_speed_per_taxi()

The first two can plug in right away to the community scatter view, the third can be used in your data question.

All three of these are actually averages of averages, because each taxi_id can have multiple trips and each pickup or dropoff area has multiple trips. So you may actually want to use a subquery. Here are two seemingly similar queries that would have different results:

```sql
SELECT
  SUM(trip_miles) / SUM(trip_hours)
FROM taxitrips
```

```sql
SELECT
  AVG(avg_speed)
FROM (
  SELECT
    trip_miles / trip_hours AS avg_speed
)
```

I think the second version (with the subquery) is the more correct metric for "average speed". The first query would be a "macro average" (all miles divided by all hours) and the second query would be a "micro average" (average of averages).